File: hackerrank-java/BalancedBrackets.java
public class BalancedBrackets {
    public static void main(String[] args) {
        
    }

    static String isBalanced(String s) {
        Stack<Character> stacky = new Stack<>();
        for (int i = 0; i < s.length(); i++) {
            if (s.charAt(i) == '{' || s.charAt(i) == '(' || s.charAt(i) == '[' ) {
                stacky.push(s.charAt(i));
            } else if (s.charAt(i) == '}' && !stacky.isEmpty()) {
                if (stacky.pop() != '{') {
                    return "NO";
                }
            } else if (s.charAt(i) == ')' && !stacky.isEmpty()) {
                if (stacky.pop() != '(') {
                    return "NO";
                }
            } else if (s.charAt(i) == ']' && !stacky.isEmpty()) {
                if (stacky.pop() != '[') {
                    return "NO";
                }
            }
        }
        return stacky.isEmpty() ? "YES" : "NO";
    }
}

File: CrackingTheCodingInterview/TechnicalQuestions/BalancedBrackets.js
function balancedBrackets(string) {
    // Write your code here.
    let stack = [];
    for (let i = 0; i < string.length; i++) {
      if (string[i] === '(' || string[i] === '[' || string[i] === '{') {
        stack.push(string[i]);
      } else if (string[i] === ']') {
        if (stack.length && stack[stack.length-1] === '[') {
          stack.pop();
        } else {
          return false;
        }
      } else if (string[i] === ')') {
        if (stack.length && stack[stack.length-1] === '(') {
          stack.pop();
        } else {
          return false;
        }
      } else if (string[i] === '}') {
        if (stack.length && stack[stack.length-1] === '{') {
          stack.pop();
        } else {
          return false;
        }
      }
    }
  
    return stack.length === 0 ? true : false;
  }